Source/AudioPlayer.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  AudioPlayer.swift
+//  Pods-VEAudioKitDemo
+//
+//  Created by Pablo Balduz on 10/07/2019.
+//
+
+import AVFoundation
+
+public protocol AudioPlayerDelegate {
+    func playerDidEnd()
+    func playerDidUpdatePosition(seconds: Float)
+}
+
+public class AudioPlayer {
+    
+    private var engine = AVAudioEngine()
+    private var player = AVAudioPlayerNode()
+    private var scheduler: Scheduler!
+    
+    private var audioFile: AVAudioFile? {
+        didSet {
+            if let audioFile = audioFile {
+                audioLengthSamples = audioFile.length
+                audioFormat = audioFile.processingFormat
+                audioSampleRate = Float(audioFormat?.sampleRate ?? 44100)
+                audioLengthSeconds = Float(audioLengthSamples) / audioSampleRate
+            }
+        }
+    }
+    
+    private var audioFormat: AVAudioFormat?
+    private var audioSampleRate: Float = 0
+    private var audioLengthSeconds: Float = 0
+    private var audioLengthSamples: AVAudioFramePosition = 0
+    
+    private var timer: Timer?
+    private var currentFrame: AVAudioFramePosition {
+        guard let lastRenderTime = player.lastRenderTime, let playerTime = player.playerTime(forNodeTime: lastRenderTime) else {
+            return 0
+        }
+        return playerTime.sampleTime
+    }
+    private var currentPosition: AVAudioFramePosition = 0
+    
+    public var delegate: AudioPlayerDelegate?
+    
+    public var audioFileURL: URL? {
+        didSet {
+            if let url = audioFileURL {
+                audioFile = try? AVAudioFile(forReading: url)
+            }
+        }
+    }
+    
+    public var isPlaying: Bool {
+        return player.isPlaying
+    }
+    
+    public var duration: Float {
+        return audioLengthSeconds
+    }
+    
+    public init() { }
```

Comment:
![Captura de pantalla 2019-07-11 a las 16 53 34](https://user-images.githubusercontent.com/9513953/61061389-774afd00-a3fc-11e9-8501-5266471d39ec.png)

happens this otherwise